src/codegen/codegen_c_visitor.cpp

```diff
@@ -1994,10 +1994,17 @@ void CodegenCVisitor::print_nmodl_constants() {
         printer->add_newline(2);
         printer->add_line("/** constants used in nmodl */");
         for (const auto& it: info.factor_definitions) {
-            std::stringstream ss;
-            ss << "static const double " << it->get_node_name() << " = ";
-            ss << it->get_value()->get_value() << ";";
-            printer->add_line(ss.str());
+#ifndef USE_LEGACY_UNITS
+#define USE_LEGACY_UNITS 0
+#endif
+#if USE_LEGACY_UNITS
+            std::string format_string = "static const double {} = {:g};";
+#else
+            std::string format_string = "static const double {} = {:.18g};";
+#endif
```

Comment:
Just a `nitpick` : you can directly use `#if defined(USE_LEGACY_UNITS)`  here?

Reply:
No because it can be defined to `0`

Reply:
Ok. I see now in CMakeLists.txt it's either 0 to 1 i.e. always defined. Why we need:
```
#ifndef USE_LEGACY_UNITS
 #define USE_LEGACY_UNITS 0
 #endif
```
?